From a2d3e43901a44cd45bc4132de024f67ca409c502 Mon Sep 17 00:00:00 2001
From: Ranbir Singh <Ranbir.Singh3@Dell.com>
Date: Tue, 24 Jan 2023 11:45:57 +0530
Subject: [PATCH 1/1] MdeModulePkg/Bus/Pci/NvmExpressPei: Fix DEADCODE Coverity
 issue

The code can reach line 65 only through the else path above at line 53.
The else path already has the same NULL check at line 55 and hence the
duplicate code lines are totally redundant which can be deleted.

Cc: Hao A Wu <hao.a.wu@intel.com>
Cc: Ray Ni <ray.ni@intel.com>
REF: https://bugzilla.tianocore.org/show_bug.cgi?id=4220
Signed-off-by: Ranbir Singh <Ranbir.Singh3@Dell.com>
---
 MdeModulePkg/Bus/Pci/NvmExpressPei/NvmExpressPeiS3.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/MdeModulePkg/Bus/Pci/NvmExpressPei/NvmExpressPeiS3.c b/MdeModulePkg/Bus/Pci/NvmExpressPei/NvmExpressPeiS3.c
index d704c62eaab3..1d03d4964025 100644
--- a/MdeModulePkg/Bus/Pci/NvmExpressPei/NvmExpressPeiS3.c
+++ b/MdeModulePkg/Bus/Pci/NvmExpressPei/NvmExpressPeiS3.c
@@ -62,10 +62,6 @@ NvmeS3SkipThisController (
     }
   }
 
-  if (S3InitDevices == NULL) {
-    return Skip;
-  }
-
   //
   // Only need to initialize the controllers that exist in the device list.
   //
-- 
2.36.1.windows.1

